### PR TITLE
scripts: west_commands: sign: Remove deprecated west.log

### PR DESCRIPTION
**Pull request by pdgendt**
The global state west.log is deprecated, replace with WestCommand logging.

### CODE REVIEW

**Review by marc-hb — DISMISSED** [review]

File: scripts/west_commands/sign.py [hunk 1]
```diff
@@ -81,18 +81,18 @@
...
 
-def config_get(west_config, section_key, fallback=None):
-    words = config_get_words(west_config, section_key)
+def config_get(cmd, section_key, fallback=None):
+    words = config_get_words(cmd, section_key)
     if words is None:
         return fallback
     if len(words) != 1:
-        log.die(f'Single word expected for: {section_key}={words}. Use quotes?')
+        cmd.die(f'Single word expected for: {section_key}={words}. Use quotes?')
```
Comment (marc-hb) [inline scripts/west_commands/sign.py:95]: Actually, `cmd` really looks like `self` now... would it make sense to move these two and make them `Forceable` methods now? If you agree but don't have time to do it, add a `TODO` suggestion in a comment?

**Review by marc-hb — DISMISSED** [review]

File: scripts/west_commands/sign.py (hunk 1 above)
Comment (pdgendt) [reply scripts/west_commands/sign.py:95]: It does make sense, I'll update.

Comment (marc-hb) [reply scripts/west_commands/sign.py]: I meant even higher to the base class, see #79678 that I just submitted.

Comment (marc-hb) [reply scripts/west_commands/sign.py]: > see https://github.com/zephyrproject-rtos/zephyr/pull/79678 that I just tested and submitted.

Merged, git conflict now, need to rebase (upvote https://github.com/orgs/community/discussions/4335 )

File: scripts/west_commands/sign.py [hunk 2]
```diff
@@ -446,9 +446,9 @@ def preprocess_toml(self, config_dir, toml_basename, subdir):
         self.command.inf(quote_sh_list(preproc_cmd))
         subprocess.run(preproc_cmd, check=True, cwd=self.build_dir)
 
-    def sign(self, command, build_dir, build_conf, formats):
-        self.command = command
-        args = command.args
+    def sign(self, cmd, build_dir, build_conf, formats):
+        self.command = cmd
```
Comment (marc-hb) [inline scripts/west_commands/sign.py:450]: This rename doubles the size of the change in this function. I agree the shorter `cmd` is long enough and better but I don't think it's so much better that it deserves that much git churn and the resulting cherry-pick/merge/rebase conflicts.

**Review by marc-hb — APPROVED** [review]

**Review by simhein — APPROVED** [review]

**Pull request merged**